File: CPP/ByRound/0624/0624_A_Save_Luke.cpp
// Problem: CF 624 A - Save Luke
// https://codeforces.com/contest/624/problem/A

/*
 * Code Purpose:
 * This program calculates the maximum time Luke can survive before the two presses
 * come together and crush him. The presses start at positions 0 and L, moving towards
 * each other with speeds v1 and v2 respectively. Luke has a width d and must choose
 * a position to maximize his survival time.
 *
 * Algorithm:
 * - The key insight is that Luke should position himself such that the distance
 *   between the presses reduces to exactly his width d.
 * - The relative speed at which the presses approach each other is v1 + v2.
 * - Time = (Initial Distance - Width) / Relative Speed = (L - d) / (v1 + v2)
 *
 * Time Complexity: O(1) - Constant time computation
 * Space Complexity: O(1) - Only a few variables used
 */

#include <stdio.h>
#include <iostream>

using namespace std;
int main() {
  long long int i, j, k;
  double d, l, v1, v2, x;
  cin >> d >> l >> v1 >> v2;
  x = (l - d) / (v1 + v2);  // Compute the time until the presses meet at distance d
  printf("%.15lf", x);
  return 0;
}


// https://github.com/VaHiX/CodeForces/

File: Python/ByTier/A/0624_A_Save_Luke.py
# Problem: CF 624 A - Save Luke
# https://codeforces.com/contest/624/problem/A

"""
Problem: Save Luke
Algorithm: Simple simulation of two approaching objects.
Time Complexity: O(1)
Space Complexity: O(1)

The problem involves two presses moving toward each other at speeds v1 and v2,
starting at positions 0 and L respectively. Luke, with width d, needs to find
the optimal position to survive as long as possible. The key insight is that
Luke dies when the distance between the presses is less than d. The maximum
survival time is when the presses are at distance d from each other, which
happens after time (L - d) / (v1 + v2), since they approach each other at
combined speed v1 + v2. Luke should choose a position where he is safe from
both directions until the presses meet.
"""

d, l, v1, v2 = map(int, input().split())
print(float((l - d) / (v1 + v2)))


# https://github.com/VaHiX/CodeForces/